includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -578,55 +448,257 @@ protected function wc_setup_wcs_tout() {
+				'description' => __( "Don't charge for shipping.", 'woocommerce' ),
+			),
+		);
+
+		$live_rate_carrier = $this->get_wcs_shipping_carrier( $country_code );
+
+		if ( false === $live_rate_carrier || ! current_user_can('install_plugins') ) {
+			unset( $shipping_methods['live_rates'] );
+		}
+
+		return $shipping_methods;
+	}
+
+	/**
+	 * Render the available shipping methods for a given country code.
+	 *
+	 * @param string $country_code
+	 * @param string $input_prefix
+	 */
+	protected function shipping_method_selection_form( $country_code, $input_prefix ) {
+		$live_rate_carrier = $this->get_wcs_shipping_carrier( $country_code );
+		$selected          = $live_rate_carrier ? 'live_rates' : 'flat_rate';
+		$shipping_methods  = $this->get_wizard_shipping_methods( $country_code );
+		?>
+		<div class="wc-wizard-shipping-method-select">
+			<select id="<?php echo esc_attr( "{$input_prefix}[method]" ); ?>" name="<?php echo esc_attr( "{$input_prefix}[method]" ); ?>" class="method wc-enhanced-select">
+			<?php foreach ( $shipping_methods as $method_id => $method ) : ?>
+				<option value="<?php echo esc_attr( $method_id ); ?>" <?php selected( $selected, $method_id ); ?>>
+					<?php echo esc_html( $method['name'] ); ?>
+				</option>
+			<?php endforeach; ?>
+			</select>
+
+			<div class="shipping-method-description">
+			<?php foreach ( $shipping_methods as $method_id => $method ) : ?>
+				<p class="<?php echo esc_attr( $method_id ); ?>" <?php if ( $method_id !== $selected ) echo 'style="display:none"'; ?>>
+					<?php echo esc_html( $method['description'] ); ?>
+				</p>
+			<?php endforeach; ?>
+			</div>
+
+			<div class="shipping-method-settings">
+			<?php foreach ( $shipping_methods as $method_id => $method ) : ?>
+				<?php if ( empty( $method['settings'] ) ) continue; ?>
+				<div class="<?php echo esc_attr( $method_id ); ?>" <?php if ( $method_id !== $selected ) echo 'style="display:none"'; ?>>
+				<?php foreach ( $method['settings'] as $setting_id => $setting ) : ?>
+					<?php $method_setting_id = "{$input_prefix}[{$method_id}][{$setting_id}]"; ?>
+					<input type="<?php echo esc_attr( $setting['type'] ); ?>" id="<?php echo esc_attr( $method_setting_id ); ?>" name="<?php echo esc_attr( $method_setting_id ); ?>" />
+					<p class="description">
+						<?php echo esc_html( $setting['description'] ); ?>
+					</p>
+				<?php endforeach; ?>
+				</div>
+			<?php endforeach; ?>
+			</div>
+		</div>
+		<?php
+	}
+
+	/**
 	 * Shipping.
 	 */
 	public function wc_setup_shipping() {
 		$dimension_unit = get_option( 'woocommerce_dimension_unit', false );
 		$weight_unit    = get_option( 'woocommerce_weight_unit', false );
+		$country_code   = WC()->countries->get_base_country();
+		$country_name   = WC()->countries->countries[ $country_code ];
+		$wcs_carrier    = $this->get_wcs_shipping_carrier( $country_code );
+
+		// TODO: determine how to handle existing shipping zones (or choose not to)
```

Comment:
Was this missed?

Reply:
Yes. Perhaps you can help make a decision on what the behavior should be here.

It's definitely a broken experience for any user re-running the setup wizard, but how common of an occurrence is that? Is this too much of an edge case to block the initial merge? _(e.g. can we do this later?)_

Reply:
@jeffstieler We do allow rerun but I think thats mainly for skipped. How about we disable the wizard by storing an option saying it was complete?

Reply:
@mikejolley the wizard already removes the `install` notice on completion. Or are you specifically referring to shipping as "complete"?

Reply:
Actually talking about disabling it/preventing accessing it again via URL. We also show a button in the help tab to re-run. This should be removed if completed.